plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/proposals/AddOutgoingStateModification.java

```diff
@@ -67,15 +78,63 @@ protected boolean check(EObject semanticObject, View view) {
 	}
 
 	@Override
-	protected void execute(EObject semanticElement, View view) {
+	protected void execute(EObject semanticElement, View sourceView) {
 		State state = (State) semanticElement;
+		
+		//add semantic state
 		State newState = SGraphFactory.eINSTANCE.createState();
-		Transition transition = SGraphFactory.eINSTANCE.createTransition();
 		state.getParentRegion().getVertices().add(newState);
+		
+		//add notation elements for state
+		Node stateNode = addStateNotations(sourceView, newState);
+		
+		//add semantic transition
+		Transition transition = SGraphFactory.eINSTANCE.createTransition();
 		transition.setSource(state);
 		transition.setTarget(newState);
 		
+		//add notation element for transition
+		ViewService.createEdge(sourceView, stateNode, transition, SemanticHints.TRANSITION, DiagramActivator.DIAGRAM_PREFERENCES_HINT);
+		
 		if (transitionSpecification != null) transition.setSpecification(transitionSpecification);
 		if (stateName != null) newState.setName(stateName);
+
+	}
+	private Node addStateNotations(View sourceView, State newState) {
+		View region = getRegionCompartmentView(sourceView);
+		Node stateNode = ViewService.createNode(region, newState, SemanticHints.STATE,
+				DiagramActivator.DIAGRAM_PREFERENCES_HINT);
+		
+		Bounds sourceBounds = null;
+		if(sourceView instanceof Node){
+			Node mynode = (Node)sourceView;
+			LayoutConstraint layoutConstraint = mynode.getLayoutConstraint();
+			if(layoutConstraint instanceof Bounds){
+				sourceBounds = (Bounds) layoutConstraint;
+			}
+		}
+		Bounds newStateBounds = NotationFactory.eINSTANCE.createBounds();
+		newStateBounds.setX(sourceBounds.getX()+200);
```

Comment:
Does not help, because CreateViewAndOptionallyelementCommand works with draw2d EditParts instead of Notation elements.
i just figured out that the element was added ~ source.x + 150 & source.y + 50 and added it hard coded into the bounds calculation of the new state view.